src/beignet/_newton_raphson.py

```diff
@@ -0,0 +1,172 @@
+import warnings
+from typing import Any, Callable, Optional, Union
+
+import torch
+from torch import Tensor
+
+# Define function signature types
+FuncType = Callable[[Tensor, Tensor], tuple[Tensor, Any]]
+FPrimeType = Callable[[Tensor, Tensor, Any], Tensor]
+
+
+def newton_raphson(
```

Comment:
A general comment, the semantics feel strange. Ideally, Newton-Raphson will respects the feasibility region or it will treat bound violations as an exception.